src/Microsoft.AspNetCore.Mvc.ViewFeatures/ViewFeatures/DefaultHtmlGenerator.cs

```diff
@@ -1239,6 +1258,11 @@ internal static object GetModelStateValue(ViewContext viewContext, string key, T
                 AddPlaceholderAttribute(viewContext.ViewData, tagBuilder, modelExplorer, expression);
             }
 
+            if (MaxLengthAttributeRenderingEnabled && _maxLengthInputTypes.Contains(suppliedTypeString))
```

Comment:
Should `_maxLengthInputTypes` be a `HashSet<>`? I'm guessing this is doing a `Enumerable.Contains`?

Reply:
I tried to be consistent with how the `_placeholderInputTypes` is defined.